src/components/TreeGrid.test.ts

```diff
@@ -102,12 +114,21 @@ describe('TreeGrid', () => {
 
     // Act
     const user = userEvent.setup();
-    await user.click(screen.getByLabelText(`expand ${tree.id}`));
-    await user.click(screen.getByLabelText(`collapse ${tree.id}`));
-    await user.click(screen.getByLabelText(`expand ${tree.id}`));
+
+    // expand
+    await user.click(screen.getByLabelText(`expand ${child2.id}`));
+    expect(screen.queryByText(child3.name)).toBeTruthy();
+
+    // collapse
+    await user.click(screen.getByLabelText(`collapse ${child2.id}`));
+    expect(screen.queryByText(child3.name)).toBeFalsy();
+
+    // expand
+    await user.click(screen.getByLabelText(`expand ${child2.id}`));
+    expect(screen.queryByText(child3.name)).toBeTruthy();
```

Comment:
The original test didn't have these asserts because there is already are tests for this above. The idea is that a test is only testing one thing at a time, if possible, and this test is to test the cache code in `TreeGrid`.

Reply:
Yes, I agree.